### PR TITLE
Add one-click installation for recommended extensions in multichannel Marketing page

### LINKED ISSUE #35578: [Enhancement]: Add one-click installation for recommended extensions in multichannel Marketing page
### Describe the solution you'd like

Project thread: pe2C5g-sS-p2

This issue is to implement one-click installation in WooCommerce plugin’s Marketing page.

In our current state, when users go to the Marketing page, they will see some recommended extensions like the following:

image

When users click on the “Get started” button, they will be directed to the product page on woocommerce.com.

The one-click installation idea is to allow the users to install and activate extensions (free and available on wordpress.org) directly in the Marketing page, without being directed to woocommerce.com. For paid extensions, they will still go through woocommerce.com.

The following demo is the existing prior art that we have in WooCommerce Home page to one-click install free extensions:

https://user-images.githubusercontent.com/417342/201713835-ce5a02cf-b786-482b-af80-4e1e6e06e0d5.mov

The outcome of this feature is to allow users to easily and quickly install the extensions and improve conversion rates for those extensions.

### Describe alternatives you've considered

None.

### Additional context

Project thread: pe2C5g-sS-p2

### PR DESCRIPTION
**Pull request by ecgan**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/35578.

This PR depends on woocommerce.com API changes in PR https://github.com/Automattic/woocommerce.com/pull/14729.

In this PR, we add one-click installation feature to recommended extensions in Marketing page (beta multichannel view only; this PR does not affect the normal Marketing page).

In the API response, recommended extensions will have a property `direct_install` to indicate whether it supports one-click installation. `direct_install` is `true` if the extension is available in the WordPress.org plugin listings; `false` otherwise.

If `direct_install` is `true`, we will display a "Install plugin" button. Clicking it will automatically install and activate the plugin. The plugin will then be removed from the "Discover more marketing tools" card and will appear in the "Installed extensions" card.

If `direct_install` is `false`, the plugin will have a "View details" button. Clicking it will navigate to the product page on WooCommerce.com. This is same as existing behavior; we just changed the button text from "Get started" to "View details". 

The code for "automatic install and activate" is based on existing prior art in the code base, e.g.:

https://github.com/woocommerce/woocommerce/blob/0b52411ff768279dc59bb5376255c584e797970c/plugins/woocommerce-admin/client/tasks/fills/Marketing/index.tsx#L131-L156

Other technical notes: 

- I introduced a new action called `removeRecommendedPlugin`. This is called after "automatic install and activate", to remove the extension from "Discover more marketing tools".
- I added new folders `hooks` and `types` for the shared hooks and types.
- `usePlugins` has been renamed to `useInstalledPlugins` for better clarity.

📹 Demo video of the feature with my voice annotation:

https://user-images.githubusercontent.com/417342/201116343-64597380-8ac3-42ff-9b00-6441a7ee3ae3.mp4

### How to test the changes in this Pull Request:

Pre-condition: 

- Make sure you have the changes in PR https://github.com/Automattic/woocommerce.com/pull/14729.
- Enable WooCommerce Multichannel Marketing experience in WooCommerce Settings > Advanced > Features (`/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`).

Test instructions:

1. Go to Marketing page at `/wp-admin/admin.php?page=wc-admin&path=%2Fmarketing`.
2. In the "Discover more marketing tools", find a plugin with "Install plugin" button, and click on the button. 
    - The button should become busy.
    - All the other "Install plugin" and "View details" buttons should become disabled. 
    - The plugin should be installed and activated automatically.
    - There should be a successful notice message in the bottom left corner.
        - If there is an error, there should be an error notice message in the bottom left corner. You can try this out by hardcoding a non-existing plugin slug in the code.
    - The plugin should then be removed from "Discover more marketing tools" card.
    - The plugin should appear in the "Installed extensions" card.
4. In the "Discover more marketing tools", find a plugin with "View details" button, and click on the button. 
    - The browser tab should navigate to the product page on WooCommerce.com

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by ecgan** [comment]
About the failing GitHub check https://github.com/woocommerce/woocommerce/actions/runs/3439480519/jobs/5736827762, the failed e2e tests are not related to the changes in this PR.

File: plugins/woocommerce-admin/client/marketing/hooks/useInstalledPlugins.ts
Comment (joshuatf) [inline plugins/woocommerce-admin/client/marketing/hooks/useInstalledPlugins.ts]: Any particular reason for this name change?  It looks like this returns more than just installed plugins and the previous name may be more accurate.

File: plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/DiscoverTools.tsx
Comment (joshuatf) [inline plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/DiscoverTools.tsx]: Nitpicky, but could we update `el` to `plugin`?  At first glance it feels like a dom element or something that's not a plugin.

File: plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/useRecommendedPlugins.ts
Comment (joshuatf) [inline plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/useRecommendedPlugins.ts]: Any purpose in assigning `getRecommendedPlugins` to a constant?  Does not seem to be a necessary change and adds time to find the which selector we're checking.

Side note and not 100% necessary to change in this PR if things are working okay, but `isResolving` is not quite as accurate as `hasFinishedResolution` for checking resolution status.

File: plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/DiscoverTools.tsx
Comment (joshuatf) [inline plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/DiscoverTools.tsx]: Thanks for moving this down here!  Using any function prefixed with `render` always feels like a code smell.

Optional, but this might even be worth moving to its own component if we're already moving this code.

File: plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/useRecommendedPlugins.ts
Comment (joshuatf) [inline plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/useRecommendedPlugins.ts]: Not saying this is wrong, just curious for the reasons behind wrapping in `useCallback`.  Does `removeRecommendedPlugin` change during the course of render?

File: plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/DiscoverTools.tsx
Comment (joshuatf) [inline plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/DiscoverTools.tsx]: Should we add tests around this change?

File: plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/DiscoverTools.tsx
Comment (joshuatf) [inline plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/DiscoverTools.tsx]: Maybe instead of introducing a new method for `removeRecommendedPlugin` we could filter this list for plugins already installed.

I'm hesitant to introduce a new method `removeRecommendedPlugin` that does not persist on page refresh.

**Review by joshuatf — COMMENTED** [review]
Thanks for the changes here, @ecgan!   Code looks good overall.  Left a few comments and questions, but nothing too major.

I think skipping the introduction of `removeRecommendedPlugin` might be a good idea if we can rely on installed plugins to hide this item from the bottom list.

File: plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/useRecommendedPlugins.ts
Comment (puntope) [reply plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/useRecommendedPlugins.ts]: > Side note and not 100% necessary to change in this PR if things are working okay, but isResolving is not quite as accurate as hasFinishedResolution for checking resolution status.

@joshuatf  Just for my curiosity, could you explain briefly the difference?

File: plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/DiscoverTools.tsx
Comment (puntope) [reply plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/DiscoverTools.tsx]: This is going to be also one of my feedback. I miss more tests for the feature.

File: plugins/woocommerce-admin/client/marketing/data/reducer.js
Comment (puntope) [inline plugins/woocommerce-admin/client/marketing/data/reducer.js]: Same as https://github.com/woocommerce/woocommerce/pull/35542/files/fac8079bf1cab3f7778c9139037f086829080e4f#r1023122631

very minor, but still nice to replace `el` with something more meaningful

File: plugins/woocommerce-admin/client/marketing/hooks/useInstalledPlugins.ts
Comment (puntope) [reply plugins/woocommerce-admin/client/marketing/hooks/useInstalledPlugins.ts]: In this one I'm going to agree with @ecgan solution since for me it seems that the return is only related to the  installed plugins. usePlugins seems more generic for me and is not really clear.

💅  If we are going to refactor it tho... besides the name it would be even more helpful to add a JSDOC at the top :)

File: plugins/woocommerce-admin/client/marketing/data/actions.js
Comment (puntope) [inline plugins/woocommerce-admin/client/marketing/data/actions.js]: In case we finally hold this action here it would be nice to add JSDOC comment at the top

File: plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/DiscoverTools.tsx
Comment (puntope) [inline plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/DiscoverTools.tsx]: A JSDOC comment at the top of this method would be cool

File: plugins/woocommerce-admin/client/marketing/hooks/useInstalledPlugins.ts
Comment (joshuatf) [reply plugins/woocommerce-admin/client/marketing/hooks/useInstalledPlugins.ts]: I think in this case, the logic in here does not just deal with "installed" plugins.  It also deals with plugins that are in the process of activation, activating plugins, and "loading" plugins after installation.  While this semantically nitpicky, the semantics here are important.

I don't want to dig into this too much in this PR since I know it was not the intention of this PR to open this can of worms, but I don't think a unique data store should have been created around plugin activation specific to marketing.  We have a [data store for plugins](https://github.com/woocommerce/woocommerce/blob/2e1a73a6dc990ba101693a5802470f68e6f9b1c5/packages/js/data/src/plugins/index.ts) in WooCommerce that would work well here.  We should strive to not reinvent the wheel and consolidate efforts behind existing packages.

File: plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/useRecommendedPlugins.ts
Comment (joshuatf) [reply plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/useRecommendedPlugins.ts]: Sure!  `isResolving` is a selector that checks the current state of a resolver, while `hasFinishedResolution` checks that a resolver has both begun and completed.

The subtle distinction is that `isResolving` for all selectors will initially be `false` until a resolver has kicked off.  This means that for a brief moment, it may appear that resolution is complete before it has even begun.  This can lead to FOUC and other unexpected behavior when we expect items to initially not be loaded.

There are more details on data store selectors in [this blog post](https://unfoldingneurons.com/2020/wordpress-data-store-properties-resolvers), but happy to answer any other questions around this that I can.

File: plugins/woocommerce-admin/client/marketing/hooks/useInstalledPlugins.ts
Comment (ecgan) [reply plugins/woocommerce-admin/client/marketing/hooks/useInstalledPlugins.ts]: > Any particular reason for this name change?

The main reason is because we already have a [`useRecommendedPlugins`](https://github.com/woocommerce/woocommerce/blob/0482860c74bffa60a11c5489b649b21796b253df/plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/useRecommendedPlugins.ts) for the "Discover more marketing tools" card. If we continue to have `usePlugin`, it feels confusing.

I renamed it to `useInstalledPlugins` because it is used in the "Installed extensions" card.

> I think in this case, the logic in here does not just deal with "installed" plugins. It also deals with plugins that are in the process of activation, activating plugins, and "loading" plugins after installation. While this semantically nitpicky, the semantics here are important.

Yeah, I thought about that too, but I couldn't think of a nice hook name for it. 

In the end, I named it `useInstalledPlugins` because it is mainly used in the "Installed extensions" card (although it is also used in the DiscoverTools component).

File: plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/DiscoverTools.tsx
Comment (ecgan) [reply plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/DiscoverTools.tsx]: > I'm hesitant to introduce a new method `removeRecommendedPlugin` that _does not persist on page refresh_.

What do you mean? 🤔 

In the code we have now, after installing a recommended plugin, if we do a page refresh, the list of recommended plugins will not have the installed plugin, due to [filtering in the back-end PHP code](https://github.com/woocommerce/woocommerce/blob/2297787a4f1541a4471c38ccdd44acc936b96bb6/plugins/woocommerce/src/Admin/API/Marketing.php#L117-L119).

`removeRecommendedPlugin` is used to remove the plugin from the recommended list before the page refresh.

File: plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/DiscoverTools.tsx
Comment (ecgan) [reply plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/DiscoverTools.tsx]: Done in [`ccc4dd4` (#35542)](https://github.com/woocommerce/woocommerce/pull/35542/commits/ccc4dd4b09e7fe7682dc638d561d909bdf66e9fa).

> At first glance it feels like a dom element or something that's not a plugin.

Kind of a personal preference or practice, I am used to writing array methods like `array.map( (el, idx, arr) => {} )`, where `el` means array element. I remember I learned these naming from MDN long time ago.

File: plugins/woocommerce-admin/client/marketing/data/reducer.js
Comment (ecgan) [reply plugins/woocommerce-admin/client/marketing/data/reducer.js]: Done in [`ccc4dd4` (#35542)](https://github.com/woocommerce/woocommerce/pull/35542/commits/ccc4dd4b09e7fe7682dc638d561d909bdf66e9fa).

File: plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/useRecommendedPlugins.ts
Comment (ecgan) [reply plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/useRecommendedPlugins.ts]: > Any purpose in assigning `getRecommendedPlugins` to a constant? Does not seem to be a necessary change and adds time to find the which selector we're checking.

When I was developing it, I had `invalidateResolution( 'getRecommendedPlugins', [category] )`, and there are multiple `'getRecommendedPlugins'` strings, so I refactor it into a constant variable. Then, I removed `invalidateResolution` but didn't change this.

I have removed the constant variable and used inline string `'getRecommendedPlugins'` in [`029d4b7` (#35542)](https://github.com/woocommerce/woocommerce/pull/35542/commits/029d4b775aae0bb0c6c061d59ac5e68d9b6c6412).

> Side note and not 100% necessary to change in this PR if things are working okay, but `isResolving` is not quite as accurate as `hasFinishedResolution` for checking resolution status.

Ahh yes, thanks for that reminder! I have fixed this in [`a3cde4b` (#35542)](https://github.com/woocommerce/woocommerce/pull/35542/commits/a3cde4b91df9704ac5aef9e5a577164b6bb41613).

File: plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/useRecommendedPlugins.ts
Comment (ecgan) [reply plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/useRecommendedPlugins.ts]: When I was developing it, without the `useCallback`, there were infinite renders and network requests. However, I just tried it again and I couldn't reproduce it anymore 🤷 😅  . I have removed it in [`ffbe20e` (#35542)](https://github.com/woocommerce/woocommerce/pull/35542/commits/ffbe20e4cacc33cbf6fb33f9979da2ffdfc76e21).

File: plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/DiscoverTools.tsx
Comment (ecgan) [reply plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/DiscoverTools.tsx]: Good point! Added in [`c649ad0` (#35542)](https://github.com/woocommerce/woocommerce/pull/35542/commits/c649ad0f8df651ac2cd12ced0660e7366192e25e).

File: plugins/woocommerce-admin/client/marketing/data/actions.js
Comment (ecgan) [reply plugins/woocommerce-admin/client/marketing/data/actions.js]: Done in [`aac6c8b` (#35542)](https://github.com/woocommerce/woocommerce/pull/35542/commits/aac6c8bffd88ac6f07f752d6ee49eaa5e1a159b7).

File: plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/DiscoverTools.tsx
Comment (ecgan) [reply plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/DiscoverTools.tsx]: Done in [`aac6c8b` (#35542)](https://github.com/woocommerce/woocommerce/pull/35542/commits/aac6c8bffd88ac6f07f752d6ee49eaa5e1a159b7).

**Comment by ecgan** [comment]
> I think skipping the introduction of `removeRecommendedPlugin` might be a good idea if we can rely on installed plugins to hide this item from the bottom list.

@joshuatf , I'm not very keen of the idea. If we have that, then imagine the following scenario:

1. We first load the page. There are 8 items in the wp-data store, and 8 items in the recommended plugin list UI.
2. We install and activate a recommended extension. We rely on installed plugins to do the filtering without `removeRecommendedPlugin`, so there are still **8 items** in the wp-data store, but **7 items** in the recommended plugin list UI.
3. We reload the page. There will be **7 items** in the wp-data store, and 7 items in the recommended plugin list UI.

I think it makes things a bit more complicated, and it would make maintainers to take more time to figure out:

- why the numbers do not match in Step 2.
- why the wp-data store item count is different in Step 2 and Step 3.

By having `removeRecommendedPlugin`, it makes the wp-data store match with the UI one to one. I think it is simpler and more straightforward.

File: plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/DiscoverTools.tsx
Comment (ecgan) [reply plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/DiscoverTools.tsx]: > Optional, but this might even be worth moving to its own component if we're already moving this code.

I actually did try that out with a new component, but by doing that, the component would then have to expose props like `onInstallAndActivate` and we have to pass things down from the parent component to the child component. The component and its API felt like not making sense to me, so I kept it this way with a flatter approach without another component level.

Comment (joshuatf) [reply plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/DiscoverTools.tsx]: Passing props down to a component is not necessarily a bad thing.  If you find yourself passing a lot of props, it may represent an underlying structural issue with the components.

That said, don't feel like this is a blocker for this PR.  The way it's done here is not bad.

**Comment by joshuatf** [comment]
>so there are still 8 items in the wp-data store, but 7 items in the recommended plugin list UI.

I don't see anything particularly wrong with this scenario and there are plenty of cases where we don't display all items returned from an endpoint in the UI.

>We reload the page. There will be 7 items in the wp-data store, and 7 items in the recommended plugin list UI.

If this is a real issue, what you're referring to here is your data store becoming out of sync with the endpoint, in which case, you can solve by 1 of 2 ways:

1. Invalidate the resolution for your endpoint after install so that it's refetched from the endpoint and updates the UI with the newly corrected values
2. Optimistically remove the item from your data store, assuming the install went well and the endpoint response will match in the future

What you appear to be attempting is option 2, however you've added the removal logic in your client-side instead of your data store.  Dispatching actions in Redux should be akin to dispatching an event.

1. [client] Dispatches install event
2. [data] Receives event and begins resolution
3. [data] Optimistically removes the item from the data store
4. [client] Receives any data changes from the data store

By using `removeRecommendedPlugin` in the consumer directly, we make the consumer directly responsible for manipulating state inside of the data store.  [We should avoid creating actions that are purely setters](https://redux.js.org/style-guide/#model-actions-as-events-not-setters).

There are quite a few cases of incorrectly using setters in this data store and across WooCommerce, but one reason I think this one may be more confusing is that someone calling `removeRecommendedPlugin` would expect that calling this continues to persist the removal upon refresh.  This is obviously not the case and the plugin would have to be installed prior to calling this in order for this to work as intended.

**Comment by ecgan** [comment]
@joshuatf , thanks for the thorough explanation! 😃 

> one reason I think this one may be more confusing is that someone calling `removeRecommendedPlugin` would expect that calling this continues to persist the removal upon refresh. This is obviously not the case and the plugin would have to be installed prior to calling this in order for this to work as intended.

Got it, that makes sense. 👍 

> Invalidate the resolution for your endpoint after install so that it's refetched from the endpoint and updates the UI with the newly corrected values

That was actually what I did in the first place, but I removed that because I thought it would be good not to make a network request just for that, and I didn't want to display the loading indicator after invalidate resolution but I couldn't figure out how to make it work at that time. 

I have removed `removeRecommendedPlugin` and used `invalidateResolution` approach in [`38abe25` (#35542)](https://github.com/woocommerce/woocommerce/pull/35542/commits/38abe25669c46e3ab8ed5a6ca1b28de1104ae2c1).

File: plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/useRecommendedPlugins.ts
Comment (joshuatf) [inline plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/useRecommendedPlugins.ts]: After install there's a very brief moment where the activated plugin remains in the list and buttons are no longer disabled.

Optionally, I think that creating separate `isLoading` and `isInitializing` states may help differentiate and disable the UI while being refetched.  Something like:

```
isInitializing: ! hasFinishedResolution( selector, [ category ] ) &&
				! plugins.length,
isLoading: ! hasFinishedResolution( selector, [ category ] );
```

**Review by joshuatf — DISMISSED** [review]
Thanks for making all the changes on this, @ecgan!  This is testing well for me.  Left one optional suggestion, but will leave that up to your discretion and pre-approving 😄 

>I didn't want to display the loading indicator after invalidate resolution but I couldn't figure out how to make it work at that time.

I think the suggestion I made on `isLoading` may help to discern between the initialization and refetching states.  But let me know if I misunderstood this!

>That was actually what I did in the first place, but I removed that because I thought it would be good not to make a network request just for that

I think the answer is that it kind of depends.  If you need the local data store to be in sync with the network (which I think you suggested by wanting the UI and data store to match) then we need to invalidate.

But you could both invalidate and remove (or filter the item based on "active" plugin status) so that it both gets removed immediately and that the data store is in sync with the server changes.

File: plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/useRecommendedPlugins.ts
Comment (ecgan) [reply plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/useRecommendedPlugins.ts]: Thanks @joshuatf , I think that's a good idea, I have made the change in [`4db827e` (#35542)](https://github.com/woocommerce/woocommerce/pull/35542/commits/4db827e70fe85be70a6259bac5c66d78371290af).

**Comment by ecgan** [comment]
> But you could both invalidate and **remove** (or filter the item based on "active" plugin status) so that it both **gets removed immediately** and that the data store is in sync with the server changes.

@joshuatf , I try to achieve the above in [`a3afcaf` (#35542)](https://github.com/woocommerce/woocommerce/pull/35542/commits/a3afcaf351345ed6a871d951bed5ac0ccbdb24c5). It is a bit similar to the previous `removeRecommendedPlugin`, but now I name it as `installAndActivateRecommendedPlugin` based on the redux style guide. Because of the change, your review approval has been dismissed. Could you have another review on this please? I wonder if you have a suggestion on the action naming?

File: plugins/woocommerce-admin/client/marketing/hooks/useInstalledPlugins.ts
Comment (ecgan) [reply plugins/woocommerce-admin/client/marketing/hooks/useInstalledPlugins.ts]: > I don't think a unique data store should have been created around plugin activation specific to marketing. We have a [data store for plugins](https://github.com/woocommerce/woocommerce/blob/2e1a73a6dc990ba101693a5802470f68e6f9b1c5/packages/js/data/src/plugins/index.ts) in WooCommerce that would work well here. We should strive to not reinvent the wheel and consolidate efforts behind existing packages.

@joshuatf , I agree with you on this, that's why in this PR, I looked into the data store for plugins and I am using the `installAndActivatePlugins` method, to use the more standardized approach in WooCommerce. 

To give a bit more context, the "unique data store specific to marketing" was created in March 2020 or earlier (see https://github.com/woocommerce/woocommerce-admin/pull/3953), while the `@woocommerce/data` package was first released in [April 2020](https://www.npmjs.com/package/@woocommerce/data/v/1.0.0), so I guess there probably wasn't a choice back then.

File: plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/DiscoverTools.tsx
Comment (ecgan) [reply plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/DiscoverTools.tsx]: This has been answered by @joshuatf in https://github.com/woocommerce/woocommerce/pull/35542#issuecomment-1317518587.

**Comment by ecgan** [comment]
On second thought, `installAndActivateRecommendedPlugin` may be confusing too like `removeRecommendedPlugin`, because someone calling `installAndActivateRecommendedPlugin` might expect it to persist but it may not be the case. 🤔

File: plugins/woocommerce-admin/client/marketing/data/actions.js
Comment (joshuatf) [inline plugins/woocommerce-admin/client/marketing/data/actions.js:128]: I agree with your above comment that this name could be confusing, but I'll leave this up to you to decide whether or not you want to change it.

I still think the easiest fix for this might be to rely on the plugins data store to see if something is being installed and activated as opposed to removing it manually.  Letting the data flow downstream (kicking off the install action and filtering in your view based on plugins that are installed/activated) is always going to be easier than handling state manually upstream (manually removing a plugin after kicking off an action in another data store).

The latter is prone to becoming out of sync and could errantly remove a recommended item when a plugin install or activation fails.

**Review by joshuatf — APPROVED** [review]
Testing well and no longer see the delay in the item being moved.

The action naming is better, but you're also right about it being possibly confusing.  Left one more comment about this, but you can decide whether or not you want to implement.  I've already held you up quite a bit on this PR 😅

File: plugins/woocommerce-admin/client/marketing/data/actions.js
Comment (ecgan) [reply plugins/woocommerce-admin/client/marketing/data/actions.js:128]: I am actually thinking about removing the marketing data store and have those marketing data (including the recommended plugins data) as local component state. That can be done in the future, possibly after Multichannel Marketing becomes the default experience and it replaces the current marketing page. When that happens, then it makes sense to have the data flow downstream and filter them in the view, and things are much simpler to see and understand without another overlapping data store. 🙂 👍

**Pull request merged**
